starport/cmd/network_chain_join.go

```diff
@@ -0,0 +1,145 @@
+package starportcmd
+
+import (
+	"fmt"
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/pkg/errors"
+	"github.com/rdegges/go-ipify"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/starport/starport/pkg/cliquiz"
+	"github.com/tendermint/starport/starport/pkg/clispinner"
+	"github.com/tendermint/starport/starport/pkg/xchisel"
+	"github.com/tendermint/starport/starport/services/network"
+)
+
+const (
+	flagGentx  = "gentx"
+	flagAmount = "amount"
+)
+
+// NewNetworkChainJoin creates a new chain join command to join
+// to a network as a validator.
+func NewNetworkChainJoin() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "join [launch-id]",
+		Short: "Join to a network as a validator by launch id",
+		Args:  cobra.ExactArgs(1),
+		RunE:  networkChainJoinHandler,
+	}
+
+	c.Flags().String(flagGentx, "", "Path to a gentx json file")
+	c.Flags().String(flagAmount, "", "If is provided sends the \"create account\" message")
+
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetHome())
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+
+	return c
+}
+
+func networkChainJoinHandler(cmd *cobra.Command, args []string) error {
+	// initialize network common methods
+	nb, s, shutdown, err := initializeNetwork(cmd)
+	if err != nil {
+		return err
+	}
+	defer shutdown()
+
+	// parse the amount
+	amountArg, _ := cmd.Flags().GetString(flagAmount)
+	amount, err := sdk.ParseCoinNormalized(amountArg)
+	if err != nil {
+		return errors.Wrap(err, "error parsing amount")
+	}
+
+	// parse launch ID
+	launchID, err := strconv.ParseUint(args[0], 10, 64)
+	if err != nil {
+		return errors.Wrap(err, "error parsing launchID")
+	}
+	if launchID == 0 {
+		return errors.New("launch ID must be greater than 0")
+	}
+
+	// parse the home path
+	home, err := getLaunchIDHome(cmd, launchID)
```

Comment:
We don't need this option IMO. Either the user provide a full path for a custom gentx through `--gentx`, either the default path is used with `~/spn/{launch-id}`